### PR TITLE
feat(axum-conversion): ported `/login` and `/signup` to axum 

### PR DESCRIPTION
**Pull request by LukeBrandon**
### Description of changes:
- [x] Added `signup` route to axum api
  - `POST /users/signup`
- [x] Added `login` route to axum api
  - `GET /users/login`
- Removed old `backend/` crate
- Renamed `backend-axum/` to `backend/`

### Misc other changes:
- Updated workflow for build
- Checked in `sqlx-data.json` to fix offline `cargo build` with `sqlx` macros
  - See: https://github.com/launchbadge/sqlx/blob/main/sqlx-cli/README.md#enable-building-in-offline-mode-with-query
  - This is still not working but it is closer.

### CODE REVIEW

**Review by FlynnOConnell — DISMISSED** [review]
I like it

**Comment by FlynnOConnell** [comment]
One day this build wont fail

File: backend-axum/src/main.rs [hunk 1]
```diff
@@ -3,22 +3,40 @@ use axum::{
...
 mod errors;
 mod models;
+mod middleware;
 
 #[tokio::main]
 async fn main() {
+
+    // RUST_LOG=backend_axum=info
+    // either in .env or before cargo run
+    dotenvy::dotenv().ok();
```
Comment (LukeBrandon) [inline backend-axum/src/main.rs]: @NeuroPyPy This is already happening, this is what the configuration module is for.

Comment (FlynnOConnell) [reply backend-axum/src/main.rs]: yea I was just testing something and forgot to remove that

**Review by FlynnOConnell — APPROVED** [review]
Looks good

**Pull request merged**